helix-term/src/commands.rs

```diff
@@ -4851,10 +4856,27 @@ fn increment_impl(cx: &mut Context, amount: i64) {
         1,
     );
 
+    let mut amount = match increment_direction {
+        IncrementDirection::Increase =>  cx.count() as i64,
+        IncrementDirection::Decrease =>  -(cx.count() as i64),
+    };
+
     let (view, doc) = current!(cx.editor);
     let selection = doc.selection(view.id);
     let text = doc.text().slice(..);
 
+
+    let increase_by = match cx.register {
+        Some(value) => match value.eq(&'#') {
+            true => match increment_direction {
+                IncrementDirection::Increase => 1,
+                IncrementDirection::Decrease => -1,
+            },
+            false => 0,
+        },
+        None => 0,
+    };
```

Comment:
This can be simplified a bit:

```rs
let sign = match increment_direction {
    IncrementDirection::Increase => 1,
    IncrementDirection::Decrease => -1,
};
let mut amount = sign * cx.count() as i64;
let increase_by = if cx.register == Some('#') {
    sign
} else {
    0
};

Reply:
It could also make sense to define IncrementDirection as

```rust
enum IncrementDirection {
    Increase = 1,
    Decrease = -1,
}
```

and then this could be:

```rust
let mut amount = increment_direction as i64 * cx.count() as i64;
let increase_by = if cx.register == Some('#') {
    increment_direction as i64
} else {
    0
};
```

but that's a bit less clear readability-wise so I don't think it's worth it

Reply:
> It could also make sense to define IncrementDirection as
> 
> ```rust
> enum IncrementDirection {
>     Increase = 1,
>     Decrease = -1,
> }
> ```
> 
> and then this could be:
> 
> ```rust
> let mut amount = increment_direction as i64 * cx.count() as i64;
> let increase_by = if cx.register == Some('#') {
>     increment_direction as i64
> } else {
>     0
> };
> ```
> 
> but that's a bit less clear readability-wise so I don't think it's worth it

I didn't even know you could use enums like that in Rust.

I updated it, how does it look now? I added comments to help readability. How does the wording look?

Reply:
Wording looks good 👍